client/scripts/views/pages/members.ts

```diff
@@ -11,51 +12,119 @@ import PageLoading from 'views/pages/loading';
 import User from 'views/components/widgets/user';
 import Sublayout from 'views/sublayout';
 import { CommunityOptionsPopover } from './discussions';
+import Compound from 'controllers/chain/ethereum/compound/adapter';
+import { sleep } from 'helpers';
+import { BigNumber } from 'ethers';
 
 interface MemberInfo {
   chain: string;
   address: string;
   count: number;
+  votes: BigNumber;
 }
 
-const MembersPage : m.Component<{}, { membersRequested: boolean, membersLoaded: MemberInfo[] }> = {
+const MEMBERS_PER_PAGE = 20;
+
+const DelegateModal: m.Component<{ address: string, name: string }, { delegateAmount: number }> = {
   view: (vnode) => {
-    const activeEntity = app.community ? app.community : app.chain;
-    if (!activeEntity) return m(PageLoading, {
-      message: 'Loading members',
-      title: [
-        'Members',
-        m(Tag, { size: 'xs', label: 'Beta', style: 'position: relative; top: -2px; margin-left: 6px' })
-      ],
-      showNewProposalButton: true,
+    return m('.DelegateModal', [
+      m('.compact-modal-title', [
+        m('h3', `Delegate to ${vnode.attrs.name ? vnode.attrs.name : 'Anonymous'}`),
+      ]),
+      m('.compact-modal-body', [
+        m('div', 'Votes to delegate'),
+        m(Input, {
+          title: 'Delegate amount',
+          oninput: (e) => {
+            const num = (e.target as any).value;
+            if (!Number.isNaN(parseInt(num, 10))) {
+              vnode.state.delegateAmount = num;
+            }
+          },
+        }),
+        m(Button, {
+          label: 'Delegate',
+          intent: 'primary',
+          onclick: async (e) => {
+            (app.chain as Compound).chain.setDelegate(vnode.attrs.address, vnode.state.delegateAmount);
+          }
+        })
+      // m(Login),
+      ]),
+    ]);
+  }
+};
+
+const MembersPage : m.Component<{}, { membersRequested: boolean, membersLoaded: MemberInfo[],
+pageToLoad: number, totalMembers: number, isCompound: boolean, voteEvents }> = {
+  oninit: (vnode) => {
+    $.get('/api/getChainEvents?type=marlin-vote-cast').then(({ result }) => {
```

Comment:
are all compound "vote-cast" emitted events typed as "marlin-vote-cast"? we should probably change this (and change our defaulting from "MPond" to "Comp"). thoughts @jnaviask?

Reply:
Hmm yeah im not sure

Reply:
Vote cast events will be emitted as `${chainId}-vote-cast` as with all other chain events.